LIS3DH.device.lib.nut

```diff
@@ -0,0 +1,486 @@
+        return _range;
+    }
+
+    // get the currently-set full-scale range of the accelerometer
+    function getRange() {
+        local range_bits = (_getReg(LIS3DH_CTRL_REG4) & 0x30) >> 4;
+        if (range_bits == 0x00) {
+            _range = 2;
+        } else if (range_bits == 0x01) {
+            _range = 4;
+        } else if (range_bits == 0x02) {
+            _range = 8;
+        } else {
+            _range = 16;
+        }
+        return _range;
+    }
+
+    // Enable/disable the accelerometer (all 3-axes)
+    function enableAccel(state = true) {
+        // LIS3DH_CTRL_REG1 enables/disables accelerometer axes
+        // bit 0 = X axis
+        // bit 1 = Y axis
+        // bit 2 = Z axis
+        local val = _getReg(LIS3DH_CTRL_REG1);
+        if (state) { val = val | 0x07; }
+        else { val = val & 0xF8; }
+        _setReg(LIS3DH_CTRL_REG1, val);
+    }
+
+    function setMode(mode) {
+        _setRegBit(LIS3DH_CTRL_REG1, 3, mode & 0x01);
+        _setRegBit(LIS3DH_CTRL_REG4, 3, mode & 0x02);
+        _mode = mode;
+    }
+
+    // Returns the deviceID (should be 51)
+    function getDeviceId() {
+        return _getReg(LIS3DH_WHO_AM_I);
+    }
+
+    function configureHighPassFilter(filters, cutoff = LIS3DH_HPF_CUTOFF1, mode = LIS3DH_HPF_DEFAULT_MODE) {
+        // clear and set filters
+        filters = LIS3DH_HPF_DISABLED | filters;
+
+        // set register
+        _setReg(LIS3DH_CTRL_REG2, filters | cutoff | mode);
+    }
+
+    //-------------------- INTERRUPTS --------------------//
+
+    // Enable/disable and configure FIFO buffer watermark interrupts
+    function configureFifoInterrupt(state, fifomode = 0x80, watermark = 28) {
+        
+        // Enable/disable the FIFO buffer
+        _setRegBit(LIS3DH_CTRL_REG5, 6, state ? 1 : 0);
+        
+        if (state) {
+            // Stream-to-FIFO mode, watermark of [28].
+            _setReg(LIS3DH_FIFO_CTRL_REG, (fifomode & 0xc0) | (watermark & 0x1F)); 
+        } else {
+            _setReg(LIS3DH_FIFO_CTRL_REG, 0x00); 
+        }
+        
+        // Enable/disable watermark interrupt
+        _setRegBit(LIS3DH_CTRL_REG3, 2, state ? 1 : 0);
+        
+    }
+
+    // Enable/disable and configure inertial interrupts
+    function configureInertialInterrupt(state, threshold = 2.0, duration = 5, 
+    options = LIS3DH_X_HIGH | LIS3DH_Y_HIGH | LIS3DH_Z_HIGH) {
+
+        // Set the enable flag
+        _setRegBit(LIS3DH_CTRL_REG3, 6, state ? 1 : 0);
+
+        // If we're disabling the interrupt, don't set anything else
+        if (!state) return;
+
+        // Clamp the threshold
+        if (threshold < 0) { threshold = threshold * -1.0; }    // Make sure we have a positive value
+        if (threshold > _range) { threshold = _range; }          // Make sure it doesn't exceed the _range
+
+        // Set the threshold
+        threshold = (((threshold * 1.0) / (_range * 1.0)) * 127).tointeger();
```

Comment:
If I'm not missing anything, there is no need to multiple by 1.0 here. `threadhold` is already double.